client/components/admin.jsx

```diff
@@ -0,0 +1,339 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-shadow */
+/* eslint-disable react/state-in-constructor */
+import Alert from 'react-s-alert';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'prop-types';
+import moment from 'moment';
+import {
+  getMovies, searchImdb, orderStock, removeMovie, addStock,
+} from '../redux/movies/actions';
+import { getUsers, toggleAdmin } from '../redux/users/actions';
+import { adminPreviousOrders } from '../redux/cart/actions';
+import { adminInventoryFilter } from '../utilities';
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Admin extends Component {
+  state = {
+    searchInput: '',
+    stockSearch: '',
+  }
+
+  async componentDidMount() {
+    // eslint-disable-next-line no-shadow
+    const {
+      getMovies, getUsers, adminPreviousOrders, movies,
+    } = this.props;
+    adminPreviousOrders();
+    getMovies();
+    getUsers();
+    const lowStock = [];
+    movies.forEach((movie) => {
+      if (movie.stock <= 10) {
+        lowStock.push(movie.title);
+      }
+    });
+    if (lowStock.length) {
+      Alert.warning(`The following movies are running low: ${lowStock.join(', ')}`, {
+        position: 'top',
+        timeout: 'none',
+      });
+    }
+  }
+
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    // eslint-disable-next-line no-shadow
+    const { searchImdb } = this.props;
+    const { searchInput } = this.state;
+    await searchImdb(searchInput);
+    this.setState({
+      searchInput: '',
+    });
+  }
+
+  handleOrder = async (e, title) => {
+    // eslint-disable-next-line no-shadow
+    const { orderStock } = this.props;
+    await orderStock(e.target.value, title);
+    // eslint-disable-next-line no-alert
+    // alert('Movie is now added to inventory!');
+  }
+
+  handleRemoveMovie = async (e, title) => {
+    // eslint-disable-next-line no-shadow
+    const { removeMovie } = this.props;
+    await removeMovie(e.target.value, title);
+    // eslint-disable-next-line no-alert
+  };
+
+  toggleAdminRole = async (userId, userName, isAdmin) => {
+    // eslint-disable-next-line no-shadow
+    const { toggleAdmin } = this.props;
+    await toggleAdmin(userId, userName, isAdmin);
+  };
+
+  render() {
+    const { searchInput, stockSearch } = this.state;
```

Comment:
Most of these .map/ .filter functions _should_ be moved out to some smaller util files.